File: src/app/dtos/order.detail.dto.ts
// import { Type } from "class-transformer";
// import { IsNotEmpty,
//     IsString,
//     IsPhoneNumber, 
//     IsDate, 
//     IsNumber,
//     ArrayMinSize,
//     ValidateNested
//    } from "class-validator";
import { Product } from "../models/product";

export class OrderDetailDTO{
    order_id: number;
    product: Product;
    price: number;
    numberOfProduct: number;
    total_money: number;
    color: string;
    
    constructor(data: any){
        this.order_id = data.order_id;
        this.product = data.product;
        this.price = data.price;
        this.numberOfProduct = data.number_of_product;
        this.total_money = data.total_money;
        this.color = data.color;

    }
}

File: src/main/java/com/project/shopapp/dtos/OrderDetailDTO.java
package com.project.shopapp.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.Min;
import lombok.*;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class OrderDetailDTO {
    @JsonProperty("order_id")
    @Min(value = 1, message = "Order's ID must be > 0")
    private long orderId;

    @JsonProperty("product_id")
    @Min(value = 1, message = "Product's ID must be > 0")
    private long productId;

    @Min(value = 0, message = "Price must be >= 0")
    private float price;

    @JsonProperty("number_of_products")
    @Min(value = 1, message = "Number of product must be > 0")
    private int numberOfProduct;

    @JsonProperty("total_money")
    @Min(value = 0, message = "Total money must be >= 0")
    private float totalMoney;

    private String color;
}
